patroni/ha.py

```diff
@@ -286,31 +291,43 @@ def demote(self, delete_leader=True):
         else:
             self.state_handler.follow(None, None)
 
-    def process_manual_failover_from_leader(self):
-        failover = self.cluster.failover
-
-        if failover.scheduled_at:
-            # If the failover is in the far future, we shouldn't do anything and just return.
+    def should_run_scheduled_action(self, action_name, scheduled_at, cleanup_fn):
+        if scheduled_at:
+            # If the faildover is in the far future, we shouldn't do anything and just return.
             # If the failover is in the past, we consider the value to be stale and we remove
             # the value.
             # If the value is close to now, we initiate the failover
             now = datetime.datetime.now(pytz.utc)
             try:
-                delta = (failover.scheduled_at - now).total_seconds()
+                delta = (scheduled_at - now).total_seconds()
 
                 if delta > self.patroni.nap_time:
-                    logging.info('Awaiting failover at %s (in %.0f seconds)', failover.scheduled_at.isoformat(), delta)
-                    return
+                    logging.info('Awaiting {0} at %s (in %.0f seconds)'.format(action_name),
+                                 scheduled_at.isoformat(), delta)
```

Comment:
Actually I am not only against mixing of `format` and printf-style, but against of using `format` with logging module (debug, info, warning, error, exception, critical).